GT6 cable quality of life improvements

mitchej123 opened the pull request:
Allow use of wire cutter/soldering item on the tile entity, as well as the wire, for a connect/disconnect. Useful for when the cable you want to power the machine is hidden behind the machine and not easily accessible.

Porting upstream from https://github.com/GTNewHorizons/GT5-Unofficial/pull/64

mitchej123 committed:
Allow use of wire cutter/soldering item on the tile entity, as well as the wire, for a connect/disconnect

Technus commented:
Just for a heads up might fuck up current behaviour of soldering iron on machine sides... 
which changes the strenght of redstone signal emmited

Technus commented:
By that i mean it overlaps with it, even if there is a check what should happen an overlapping behaviour is bad imo.

Just a matter of preference.

Technus commented:
Other idea would be to implement other tool for "electical" wire soldering/cutting. 

like electric wire cramping/cutting tool

mitchej123 committed:
revert soldering iron changes

mitchej123 commented:
Reverted the soldering iron changes; left the wire cutter right click behavior.

Technus committed:
Added missing overlays, made the passtrough of wire cutting/connectiong trough gt machine blocks work only if the player is isneaking. And because of that, added back the soldering iron support.

Technus commented:
@mitchej123 cherrypick this commit to that merge branch
https://github.com/GTNewHorizons/GT5-Unofficial/commit/c89acf537d0b729f41147f8575016d96f53c6ee2

mitchej123 committed:
Solar Panel Fix

Two main pieces:
1) Updates solar panel (cover) behavior to always show connected, and
allows connect() to work without a tile entity.  This bit is mostly
optional and mainly included so the client renders the panels as
connected.

2) Updates injectEnergy() to allow injection from unconnected, but
UNKNOWN directions (6) - which is what solar panels and transformers
use.

mitchej123 commented:
Included a fix for an issue with gt6 style wires and GT solar panels (covers) that some players noticed.  CC 

CC @Antifluxfield

mitchej123 commented on src/main/java/gregtech/api/metatileentity/implementations/GT_MetaPipeEntity_Cable.java:
```diff
@@ -269,29 +275,61 @@ public int connect(byte aSide) {
+                }
+            }
+            
+            boolean sHasSolarPanel = coverBehavior instanceof GT_Cover_SolarPanel;
```
Mentioned downstream - I don't love this; but I'm not aware of any other "covers" that inject energy. Also it's the only connection that doesn't require a tile entity on the other side.

mitchej123 committed:
Make AE2 GT-p2p tunnels connect to gt cables

mitchej123 commented:
Added a fix for cables & AE2 GT EU p2p tunnels
